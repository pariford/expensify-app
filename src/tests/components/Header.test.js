//react-test-renderer -> a new jest library to test components

import ReactShallowRenderer from "react-test-renderer/shallow";
import React from "react";
import { shallow } from "enzyme";
import Header from "../../components/HeaderPage";

test("should render the header page correctly", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper.find("h1").text()).toBe("Expensify");
  expect(wrapper).toMatchSnapshot();
  //Best way to do is using enzyme, which is above
  /* const renderer = new ReactShallowRenderer();
  renderer.render(<Header />);
  expect(renderer.getRenderOutput()).toMatchSnapshot(); */
});
