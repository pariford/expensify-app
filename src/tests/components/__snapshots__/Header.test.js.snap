// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the header page correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="selected"
    aria-current="page"
    exact={true}
    to="/"
  >
    Home Page
  </NavLink>
  <NavLink
    activeClassName="selected"
    aria-current="page"
    exact={true}
    to="/create"
  >
    Add Component
  </NavLink>
  <NavLink
    activeClassName="selected"
    aria-current="page"
    exact={true}
    to="/edit"
  >
    Edit Component
  </NavLink>
  <NavLink
    activeClassName="selected"
    aria-current="page"
    exact={true}
    to="/help"
  >
    Help Page
  </NavLink>
</header>
`;
